### ISSUE TITLE
Implement NT-Xent loss (implicit negative samples)

### ISSUE THREAD

**Issue by ziw-liu** [mehta-lab/VisCy#136]
Implement InfoNCE-style (e.g. NT-Xent from SimCLR) training objective for contrastive learning. This changes sampling strategy so it probably should be done with #123.

**Comment by mattersoflight** [comment]
@ziw-liu as discussed, first implement NT-Xent loss for the "classical sampling strategy", which can be done with the current dataloader.

Based on the results, we can figure out our next steps.

**Comment by ziw-liu** [comment]
Just to log our thoughts, a 'proper' implementation would be to treat each track as a sample instead of each node as a sample in the dataset, so that each track contributes one positive pair per epoch, and no batch will contain more than 2 patches from the same track.

**Comment by ziw-liu** [comment]
@edyoshikun See also official [tensorflow implementation](https://github.com/google-research/simclr/blob/383d4143fd8cf7879ae10f1046a9baeb753ff438/objective.py#L34) and a pytorch [reproduction](https://github.com/sthalles/SimCLR/blob/1848fc934ad844ae630e6c452300433fe99acfd9/simclr.py#L26).
